marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -261,8 +268,9 @@ handleAction (PickupAction pickupAction) = case pickupAction of
           handleAction AddNewWallet
           handleAction $ PickupAction $ Pickup.PickupWallet walletDetails
   Pickup.PickupWallet walletDetails -> do
-    -- TODO: set some state flag to "picking up..." so that we can notify the user in the view
-    -- that something is happening (because all these requests may take a while)
+    -- Picking up a wallet takes a while (several trips to the PAB are required). So first of all we set
+    -- the pickingUp flag to true, so that we can inform the user that this is all ongoing...
+    assign (_pickupState <<< _pickingUp) true
```

Comment:
In a previous PR I saw you changed from RemoteData to an Either, but now we are introducing a boolean to indicate loading... I'm wondering if we should use RemoteData back again.

Reply:
I was wondering that too, but it wasn't obvious to me which property should be RemoteData in this case. I'll give it some more thought for the next PR.